pkg/operator/ceph/cluster/mon/drain.go

```diff
@@ -0,0 +1,103 @@
+/*
+Copyright 2021 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package mon
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	"github.com/rook/rook/pkg/operator/k8sutil"
+	policyv1beta1 "k8s.io/api/policy/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+const (
+	monPDBName = "rook-ceph-mon-pdb"
+)
+
+func (c *Cluster) reconcileMonPDB() error {
+	monCount := c.spec.Mon.Count
+	if monCount <= 2 {
+		logger.Debug("managePodBudgets is set, but mon-count <= 2. Not creating a disruptionbudget for Mons")
+		return nil
+	}
+
+	op, err := c.createOrUpdateMonPDB()
+	if err != nil {
+		return errors.Wrapf(err, "failed to reconcile mon pdb on op %q", op)
+	}
+	return nil
+}
+
+func (c *Cluster) createOrUpdateMonPDB() (controllerutil.OperationResult, error) {
```

Comment:
What if this method took the maxUnavailable as a param, then everyone could just call this?
```suggestion
func (c *Cluster) createOrUpdateMonPDB(maxUnavailable int) (controllerutil.OperationResult, error) {
```

Reply:
done.